prow/github/client.go

```diff
@@ -524,8 +548,8 @@ func (s *addHeaderTransport) RoundTrip(r *http.Request) (*http.Response, error)
 }
 
 // NewClient creates a new fully operational GitHub client.
-func NewClient(getToken func() []byte, censor func([]byte) []byte, graphqlEndpoint string, bases ...string) Client {
-	return NewClientWithFields(logrus.Fields{}, getToken, censor, graphqlEndpoint, bases...)
+func NewClient(getToken func() []byte, censor func([]byte) []byte, appOptions *GitHubAppOptions, graphqlEndpoint string, bases ...string) (Client, error) {
```

Comment:
Alternatively I could also add a new constructor for github apps auth, that would require less changes